packages/api/src/controllers/clip.ts

```diff
@@ -0,0 +1,154 @@
+import { validatePost } from "../middleware";
+import { Request, Router } from "express";
+import _ from "lodash";
+import { db } from "../store";
+import { NotFoundError } from "../store/errors";
+import { pathJoin } from "../controllers/helpers";
+import {
+  createAsset,
+  validateAssetPayload,
+  defaultObjectStoreId,
+  catalystPipelineStrategy,
+} from "./asset";
+import { generateUniquePlaybackId } from "./generate-keys";
+import { v4 as uuid } from "uuid";
+import { DBSession } from "../store/db";
+import { fetchWithTimeout } from "../util";
+import { DBStream } from "../store/stream-table";
+
+const app = Router();
+
+app.post("/", validatePost("clip-payload"), async (req, res) => {
+  const playbackId = req.body.playbackId;
+  const userId = req.user.id;
+
+  const id = uuid();
+  let uPlaybackId = await generateUniquePlaybackId(id);
+
+  const content = await db.stream.getByPlaybackId(playbackId); //||
+  //(await db.asset.getByPlaybackId(playbackId));
+
+  let isStream: boolean;
+  if (content && "streamKey" in content) {
+    isStream = true;
+  }
+
+  if (!content) {
+    throw new NotFoundError("Content not found");
+  }
+
+  const user = await db.user.get(content.userId);
+
+  if (!user || userId !== content.userId) {
+    throw new NotFoundError("Content not found");
+  }
+
+  if ("suspended" in content && content.suspended) {
+    throw new NotFoundError("Content not found");
+  }
+
+  let asset = await validateAssetPayload(
+    id,
+    uPlaybackId,
+    content.userId,
+    Date.now(),
+    defaultObjectStoreId(req),
+    req.config,
+    {
+      name: req.body.name || `clip_${uPlaybackId}`,
```

Comment:
supernit: for consistency with [the assets from recording](https://github.com/livepeer/livepeer-com/blob/4a3856a0e7efdf4fa6ecf9505c43062dc5301ecd/packages/api/src/webhooks/cannon.ts#L558), lets use dash here
```suggestion
      name: req.body.name || `clip-${uPlaybackId}`,
```